sklearn/tree/_criterion.pyx

```diff
@@ -230,10 +261,19 @@ cdef class ClassificationCriterion(Criterion):
         self.start = 0
         self.pos = 0
         self.end = 0
+        self.missing_direction = 2  # Ignore the missing values by default
 
         self.n_outputs = n_outputs
+
+        self.n_missing = 0
+        self.weighted_n_missing = 0
+
+        self.n_nonmissing = 0
+        self.weighted_n_nonmissing = 0
```

Comment:
I'm not a huge fan of 'nonmissing'. Couldn't you just do the weight of the valued samples using weighted_n_node_samples, and the weight of the missing samples using weighted_n_missing?